frontend/src/tests/ddah-tests.js

```diff
@@ -414,5 +507,57 @@ export function ddahsEmailAndDownloadTests(api) {
 
     // For this test, it is sufficient to only check the HTML version,
     // which can be downloaded by leaving off the `.pdf` suffix from the public route.
-    it.todo("Downloaded ddah includes all duties (from all categories)");
+    it("Downloaded ddah includes all duties (from all categories)", async () => {
+        const ddahWithAllDutyTypes = {
+            assignment_id: assignments[0].id,
+            duties: [
+                {
+                    order: 2,
+                    hours: 25,
+                    description: "marking:Marking the test",
+                },
+                {
+                    order: 1,
+                    hours: 4,
+                    description: "training:TA training",
+                },
+                {
+                    order: 3,
+                    hours: 40,
+                    description: "contact:Tutorials",
+                },
+                {
+                    order: 5,
+                    hours: 40,
+                    description: "note:Some note",
+                },
+                {
+                    order: 6,
+                    hours: 40,
+                    description: "meeting:Weekly meetings",
+                },
+                {
+                    order: 7,
+                    hours: 40,
+                    description: "prep:Assignment prep",
+                },
+                {
+                    order: 4,
+                    hours: 40,
+                    description: "other:Other duties",
+                },
+            ],
+        };
+
+        let resp = await apiPOST(`/admin/ddahs`, ddahWithAllDutyTypes);
+        expect(resp).toHaveStatus("success");
+
+        const ddahHtml = (
+            await axios.get(
+                `${BACKEND_BASE_URL}/public/ddahs/${ddah.url_token}`
+            )
+        ).data;
+
+        expect(ddahHtml).toMatchSnapshot();
```

Comment:
Opted for a snapshot here to make sure this test fails every time HTML is updated - this way the snapshot will always be checked for containing the categories

Reply:
I don't like this. It's not going to be reliable, and the html will likely change often. Please check the html for the duties.

Reply:
The reason why I think it is better as a snapshot is because we want to make sure all categories are properly displayed, not just contained in the HTML. I can change it to a simple matcher, but if the styles make them not render properly or HTML changes -  I thought we would want to revisit the snapshot and make sure categories are there. We could possibly use Jest's snapshot serializer to display a more meaningful message.

For the matcher option, would you like a simple `.stringContaining()` or anything else for the check? :slightly_smiling_face:

Reply:
`stringContaining` would be fine. We don't want to reply on by-hand inspection. That's a recipe for disaster.